### PR TITLE
Use Prisma's programmatic API in keystone-next reset and make keystone-next reset actually work

### PR DESCRIPTION
**Pull request by mitchellhamilton**
Same as #5148 but for reset. This also makes the reset command actually runnable, it wasn't in the cli file before.

### CODE REVIEW

File: packages/adapter-prisma/src/adapter-prisma.js [hunk 1]
```diff
@@ -269,20 +274,18 @@ class PrismaAdapter extends BaseKeystoneAdapter {
...
-        this._runPrismaCmd(`migrate reset --force --preview-feature`);
-        await runPrototypeMigrations(this._url(), this.prismaSchema, path.resolve(this.schemaPath));
-      }
-    } else if (this.provider === 'sqlite') {
-      const tables = await this.prisma.$queryRaw(
-        "SELECT name FROM sqlite_master WHERE type='table';"
-      );
-      for (const { name } of tables) {
-        await this.prisma.$queryRaw(`DELETE FROM "${name}";`);
+        throw new Error('Only "postgresql" and "sqlite" providers are supported');
```
Comment (mitchellhamilton) [inline packages/adapter-prisma/src/adapter-prisma.js:285]: Given we need to have a case for all the providers we'll want to support to run the tests fast enough, having a general case that we never actually use or test seems silly so I changed to an error.

File: packages/adapter-prisma/src/migrations.ts [hunk 2]
```diff
@@ -103,6 +103,30 @@ ${chalk.greenBright('All migrations have been successfully applied.')}`);
...
 
+// https://github.com/prisma/prisma/blob/527b6bd35e7fe4dbe854653f872a07b25febeb65/src/packages/migrate/src/commands/MigrateReset.ts
+export async function resetDatabaseWithMigrations(dbUrl: string, schemaPath: string) {
+  await ensureDatabaseExists(dbUrl, path.dirname(schemaPath));
+  const migrate = new Migrate(schemaPath);
+
+  try {
+    await migrate.reset();
+
+    const { appliedMigrationNames: migrationIds } = await migrate.applyMigrations();
```
Comment (timleslie) [inline packages/adapter-prisma/src/migrations.ts:114]:
```suggestion
    await runMigrateWithDbUrl(dbUrl, () => migrate.reset());

    const { appliedMigrationNames: migrationIds } = await runMigrateWithDbUrl(dbUrl, () =>
      migrate.applyMigrations()
    );
```

**Review by timleslie — CHANGES_REQUESTED** [review]

File: packages/adapter-prisma/src/adapter-prisma.js (hunk 1 above)
Comment (timleslie) [reply packages/adapter-prisma/src/adapter-prisma.js:285]: Oh, nice catch, this was causing our sqlite tests to be way slower than they needed to be on CI!

**Review by timleslie — APPROVED** [review]
Looks good 👍

**Pull request merged**